src/io.rs

```diff
@@ -208,6 +321,20 @@ pub struct ByteLines<B> {
     buf: B,
 }
 
+/// An iterator over records from an instance of
+/// [`std::io::BufRead`](https://doc.rust-lang.org/std/io/trait.BufRead.html).
```

Comment:
Could this add a sentence explaining what a byte record is? e.g., "A byte record is any sequence of bytes terminated by a particular byte chosen by the caller. For example, NUL separate byte strings are said to be NUL-terminated byte records."